WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrderViewModel.swift

```diff
@@ -220,6 +225,30 @@ extension NewOrderViewModel {
             self.quantity = quantity
         }
     }
+
+    /// Representation of customer data display properties
+    ///
+    struct CustomerDataViewModel {
+        let isDataAvailable: Bool
+        let fullName: String?
+        let email: String?
+        let billingAddressFormatted: String?
+        let shippingAddressFormatted: String?
+
+        init(billingAddress: Address?, shippingAddress: Address?) {
+            if billingAddress == nil, shippingAddress == nil {
+                isDataAvailable = false
+            } else {
+                isDataAvailable = true
+            }
+
+            fullName = billingAddress?.fullName ?? shippingAddress?.fullName
+            email = billingAddress?.hasEmailAddress == true ? billingAddress?.email : nil
+
+            billingAddressFormatted = billingAddress?.fullNameWithCompanyAndAddress
+            shippingAddressFormatted = shippingAddress?.fullNameWithCompanyAndAddress
+        }
```

Comment:
Should this logic have some kind of unit tests?

Reply:
Totally! Added in 2db26010ad3a956877dcc1f6396a548f904b0ccc.